src/controllers/project.js

```diff
@@ -0,0 +1,113 @@
+const Project = require('../models/project');
+
+// eslint-disable-next-line consistent-return
+exports.getAllProjects = async (_, res) => {
+  try {
+    const projects = await Project.find();
+    if (projects.length === 0) {
+      res.json(projects);
+      return projects;
+    }
+  } catch (err) {
+    res.status(422).json({ message: err.message });
+  }
+};
+
+// eslint-disable-next-line consistent-return
+exports.getOneProject = async (req, res) => {
+  const { id } = req.params;
+  try {
+    const project = await Project.findById(id);
+    if (!project) {
+      res
+        .status(422)
+        .json({ message: "the project you are looking for wasn't found" });
+      return "the project you are looking for wasn't found";
+    }
+    res.json(project);
+    return project;
+  } catch (err) {
+    res.status(422).json({ message: err.message });
+    return err.message;
+  }
+};
+
+// eslint-disable-next-line consistent-return
+exports.addProject = async (req, res) => {
+  const projectData = req.body;
+  try {
+    const newProject = await Project.create(projectData);
+    res.status(201).json(newProject);
+    return projectData;
+  } catch (err) {
+    res.status(422).json({ message: err.message });
+    return err.message;
+  }
+};
+
+// eslint-disable-next-line consistent-return
+exports.filterProjects = async (req, res) => {
+  const { creator, location } = req.query;
+  if (!creator && !location) {
+    res
+      .status(400)
+      .json({ message: 'make sure you send a valid query parameter' });
+  } else {
+    const query = {};
+    if (creator) query.creator = creator;
+    if (location) query.location = location;
+    try {
+      const projects = await Project.find(query);
+      if (projects.length === 0) {
+        res.json(projects);
+        return projects;
+      }
```

Comment:
```suggestion
    res.json(projects);
```

The length check is not required, then response will be sent only when there are no projects.